### PR TITLE
Feature/tag retries

### PR DESCRIPTION
**Pull request by evansde77**
@appeltel 

have a case where push tags is being flaky. Since we already retry flaky pushes, made it retry this as well.

### CODE REVIEW

File: src/cirrus/github_tools.py [hunk 1]
```diff
@@ -269,7 +269,25 @@ def tag_release(self, tag, master='master', push=True):
...
+            while count < attempts:
+                error_flag = None
+                try:
+                    self.repo.remotes.origin.push(self.repo.head, tags=True)
+                    break
+                except Exception as ex:
+                    msg = "Error pushing tags: {}".format(ex)
+                    LOGGER.info(msg)
+                    error_flag = ex
+                    time.sleep(cooloff)
```
Comment (ksnavely) [inline src/cirrus/github_tools.py]: If this is a cool off, don't you want to scale with attempts too? 2s, 4s, 8s, that sort of thing?

File: src/cirrus/github_tools.py [hunk 2]
```diff
@@ -269,7 +269,25 @@ def tag_release(self, tag, master='master', push=True):
...
+            count = 0
+            error_flag = None
+            while count < attempts:
+                error_flag = None
+                try:
+                    self.repo.remotes.origin.push(self.repo.head, tags=True)
+                    break
+                except Exception as ex:
+                    msg = "Error pushing tags: {}".format(ex)
+                    LOGGER.info(msg)
```
Comment (ksnavely) [inline src/cirrus/github_tools.py]: Perhaps a warning instead of info?

File: src/cirrus/github_tools.py (hunk 1 above)
Comment (evansde77) [inline src/cirrus/github_tools.py]: nah, its really just to get a small window to pass some network weather or a server being a dick. Most of the examples I have seen of this failing re-run almost as fast as I can restart the build.

File: src/cirrus/release.py [hunk 3]
```diff
@@ -651,7 +651,12 @@ def merge_release(opts):
                 cooloff=rel_conf['push_retry_cooloff']
             )
             LOGGER.info(u"Tagging {} as {}".format(master, tag))
-            ghc.tag_release(tag, master)
+            ghc.tag_release(
+                tag,
+                master,
+                attempts=rel_conf['push_retry_attempts'],
```
Comment (ksnavely) [inline src/cirrus/release.py]: Worth a .get(..., 1) attempts and .get(..., 0) cooloff?

Comment (evansde77) [inline src/cirrus/release.py]: everything gets safely defaulted in the release_config method

**Comment by ksnavely** [comment]
LGTM

File: src/cirrus/github_tools.py (hunk 2 above)
Comment (evansde77) [inline src/cirrus/github_tools.py]: ok

File: src/cirrus/release.py (hunk 3 above)
Comment (ksnavely) [inline src/cirrus/release.py]: nice

**Comment by dmannarino** [comment]
LGTM

**Pull request merged**